rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+
+The vocabulary change is that the `unevaluatedProperties` keyword will only be applicable at top-level schemas (i.e. subtypes or schemas that don't specify `allOf`), and ignored when present on a schema that resides in the `allOf` array. Instead of the default value being `{ "unevaluatedProperties": {} }`, the default value will be `{ "unevaluatedProperties": false }`. Together this makes it so that all supertypes will be able to compose, as their individual, implicit `unevaluatedProperties` values won't have any effect on validation and defer evaluation to the extending type (the subtype).
+
+Concretely, this means that for a free-standing type that doesn't extend any other type such as this `Person` type:
+
+```json
+{
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/person/v/2",
+  "type": "object",
+  "title": "Person",
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/age": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/age/v/1"
+    }
+  },
+  "required": [
+    "https://blockprotocol.org/@alice/property-type/name",
+    "https://blockprotocol.org/@alice/property-type/age"
+  ]
+}
+```
+
+would implicitly have `{ "unevaluatedProperties": false }` set. In the case of the `Employee` type:
+
+```json
+{
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/employee/v/2",
+  "type": "object",
+  "title": "Employee",
+  "allOf": [
+    { "$ref": "https://blockprotocol.org/@alice/entity-type/person/v/2" }
+  ],
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/occupation": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/occupation/v/1"
+    }
+  },
+  "required": [
+    "https://blockprotocol.org/@alice/property-type/name",
+    "https://blockprotocol.org/@alice/property-type/occupation"
+  ]
+}
+```
+
+the resolved schema residing at `{ "$ref": "https://blockprotocol.org/@alice/entity-type/person/v/2" }` would _not_ have `{ "unevaluatedProperties": false }` set. but the top-level entity type would have `{ "unevaluatedProperties": false }` set.
+
+## Detecting cycles
+
+> 💭 Because of versioning mechanics in the type system added in the [versioning RFC](./0408-versioning-types.md), type versions are not able to make out proper dependency cyles, and would not allow cyclic type hierachy to exist in the literal sense as updates to types would always result in a new version identifier. Instead cycles in the below explanation is more about indirection and obfuscation of types.
```

Comment:
```suggestion
> 💭 Because of versioning mechanics in the type system added in the [versioning RFC](./0408-versioning-types.md), type versions are not able to make out proper dependency cycles, and would not allow cyclic type hierarchy to exist in the literal sense as updates to types would always result in a new version identifier. Instead cycles in the below explanation is more about indirection and obfuscation of types.
```

Reply:
I think the paragraph is also generally unclear, I'm not entirely sure what you're talking about here.

Reply:
Adjusted in 25737f66e415610a752a93d467bf8eee045386a8